From be6314eabf4b9f13df15d0f5964cf90ee1845255 Mon Sep 17 00:00:00 2001
From: Christoph Bumiller <christoph.bumiller@speed.at>
Date: Mon, 27 May 2013 14:48:47 +0200
Subject: [PATCH 11/34] gallium/draw: support hack to disable clipping (v2)

v2: drop ifdef change, which may be send later in separate commit
---
 src/gallium/auxiliary/draw/draw_context.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/gallium/auxiliary/draw/draw_context.c b/src/gallium/auxiliary/draw/draw_context.c
index b0f4ca2..20dea66 100644
--- a/src/gallium/auxiliary/draw/draw_context.c
+++ b/src/gallium/auxiliary/draw/draw_context.c
@@ -378,6 +378,11 @@ void draw_set_viewport_states( struct draw_context *draw,
        viewport->translate[1] == 0.0f &&
        viewport->translate[2] == 0.0f &&
        viewport->translate[3] == 0.0f);
+
+   draw->driver.bypass_clip_xy = vps[0].scale[3] == 0.0f;
+   draw->clip_xy = !draw->driver.bypass_clip_xy;
+   draw->identity_viewport =
+      draw->identity_viewport || vps[0].scale[3] == 0.0f;
 }
 
 
-- 
2.1.3

